x/gov/client/rest/grpc_query_test.go

```diff
@@ -0,0 +1,443 @@
+	govtestutil "github.com/cosmos/cosmos-sdk/x/gov/client/testutil"
+	"github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/suite"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	cfg := network.DefaultConfig()
+	cfg.NumValidators = 1
+
+	s.cfg = cfg
+	s.network = network.New(s.T(), cfg)
+
+	_, err := s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+
+	val := s.network.Validators[0]
+
+	// create a proposal with deposit
+	_, err = govtestutil.MsgSubmitProposal(val.ClientCtx, val.Address.String(),
+		"Text Proposal 1", "Where is the title!?", types.ProposalTypeText,
+		fmt.Sprintf("--%s=%s", cli.FlagDeposit, sdk.NewCoin(s.cfg.BondDenom, types.DefaultMinDepositTokens).String()))
+	s.Require().NoError(err)
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+
+	// vote for proposal
+	_, err = govtestutil.MsgVote(val.ClientCtx, val.Address.String(), "1", "yes")
+	s.Require().NoError(err)
+
+	// create a proposal without deposit
+	_, err = govtestutil.MsgSubmitProposal(val.ClientCtx, val.Address.String(),
+		"Text Proposal 2", "Where is the title!?", types.ProposalTypeText)
+	s.Require().NoError(err)
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TestGetProposalGRPC() {
+	val := s.network.Validators[0]
+
+	testCases := []struct {
+		name   string
+		url    string
+		expErr bool
+	}{
+		{
+			"empty proposal",
+			fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals/%s", val.APIAddress, ""),
+			true,
+		},
+		{
+			"get non existing proposal",
+			fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals/%s", val.APIAddress, "10"),
+			true,
+		},
+		{
+			"get proposal with id",
+			fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals/%s", val.APIAddress, "1"),
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			resp, err := rest.GetRequest(tc.url)
+			s.Require().NoError(err)
+
+			var proposal types.QueryProposalResponse
+			err = val.ClientCtx.JSONMarshaler.UnmarshalJSON(resp, &proposal)
+
+			if tc.expErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NotNil(proposal.Proposal)
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetProposalsGRPC() {
+	val := s.network.Validators[0]
+
+	testCases := []struct {
+		name    string
+		url     string
+		headers map[string]string
+		expErr  bool
+	}{
+		{
+			"get proposals with height 1",
+			fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals", val.APIAddress),
+			map[string]string{
+				grpctypes.GRPCBlockHeightHeader: "1",
+			},
+			true,
+		},
+		{
+			"valid request",
+			fmt.Sprintf("%s/cosmos/gov/v1beta1/proposals", val.APIAddress),
+			map[string]string{},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
if you don't execute tests in parallel, then this line is not needed. 

BTW, does our Test `Suite` support that? [Example](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks).

Reply:
It's a linter thing. Please keep this.